lesson01.md

```diff
@@ -1 +1,423 @@
+
+---
+
+### Tools
+
+We'll be using these tools during our course:
+
+* [Chrome Browser](https://www.google.com/chrome/)
+* [Visual Studio Code](https://code.visualstudio.com/)
+* [MDN Web Documentation](https://developer.mozilla.org/en-US/)
+
+Please install [Visual Studio Code](https://code.visualstudio.com/) for the next lesson
+
+---
+
+### Tools: Chrome Browser
+
+* https://www.google.com/chrome/
+* Most common browser
+* Most up to date
+* Excellent Development Tools (F12)
+
+---
+
+### Tools: Visual Studio Code
+
+* https://code.visualstudio.com/
+* Open source IDE (**I**ntegrated **D**evelopment **E**nvironment)
+* Great extensions
+* Good defaults
+* Fast
+
+---
+
+### Tools: MDN Web Documentation
+
+* https://developer.mozilla.org/en-US/
+* Up to date reference for JS
+* Good guides
+* Free
+
+---
+
+### Additional Material
+
+* [javascript.info](https://javascript.info/) - tutorials with nice (often visual) explanations
+* [freecodecamp.org](https://www.freecodecamp.org/) - full free courses on many topics
+* [udemy.com](https://www.udemy.com): Introduction to JS (course)
+* [codeacademy.com](https://www.codecademy.com): Introduction to JS (another course)
+* [edabit.com](https://edabit.com): Learn JavaScript with interactive challenges and external resources
+* [learnjavascript.online](https://learnjavascript.online): very good online course, but costs 40€
+
+---
+
+### The standard
+
+* Standardized Specification of JavaScript
+* Very technical, not good for learning
+* [http://www.ecma-international.org/ecma-262/10.0/index.html](http://www.ecma-international.org/ecma-262/10.0/index.html)
+
+---
+
+### Names of Special Characters
+
+|||
+| --- | --- |
+| `(` Parentheses `)` | `<` Angle brackets `>` |
+| `{` Braces, or curly braces `}` | `&` Ampersand `&` |
+| `[` Brackets, or square brackets `]` | <code>&#124;</code> Vertical bar, or pipe <code>&#124;</code> |
+| `;` Semicolon `;` | `:` Colon `:`  |
+| `'` Single quote `'`| `#` Hash or number sign `#` |
+| `"` Double quote `"` | `` ` `` Back tick `` ` `` |
+| `_` Underscore `_` | `*` Asterisk `*` |
+| `/` Slash, or Forward slash `/` | `~` Tilde `~` |
+| `\` Backslash `\` | `^` Caret or circumflex `^` |
+
+---
+
+<!-- .slide: id="basic-data-types" -->
+
+# Basic Data Types
+
+---
+
+### Let's start with JavaScript!
+
+* Open Google Chrome
+* Hit `F12` key
+* Click on **Console**
+
+---
+
+### Enter Some Data
+
+* Try entering your name
+<!-- .element: class="fragment" data-fragment-index="1" -->
+* try entering your shoe size
+<!-- .element: class="fragment" data-fragment-index="1" -->
+* or anything else you want, really
+<!-- .element: class="fragment" data-fragment-index="1" -->
+
+What did you try? What did you see?
+<!-- .element: class="fragment" data-fragment-index="2" -->
+
+---
+
+### Data Types
+
+* JavaScript seems to be picky about the data we enter
+* In order to understand how to enter data, we need to know about **Data Types**
+
```

Comment:
* A data type is defined by two things: a set of values and operators that can be used on the values of the set. i.e `2` and `-3` are example values from the numbers set and `+` is an operator we can use to perform an addition on `2` and `-3`. When we say Number data type, we mean all the values in numbers set plus all the meaningful operations we can perform with the values from the numbers set.

Reply:
Added in a new slide below